script/texdoclib-cli.tlu

```diff
@@ -98,52 +102,31 @@ local function parse_options()
             curr_arg = '--' .. k
         end
 
-        -- action
-        if (curr_arg == '-h') or (curr_arg == '--help') then
-            return true, 'help', cl_config
-        elseif (curr_arg == '-V') or (curr_arg == '--version') then
-            return true, 'version', cl_config
-        elseif (curr_arg == '-f') or (curr_arg == '--files') then
-            return true, 'files', cl_config
-        elseif curr_arg == '--just-view' then
-            return true, 'view', cl_config
-
-        -- mode
-        elseif (curr_arg == '-w') or (curr_arg == '--view') then
-            insert_cl_config('mode', 'view', curr_arg)
-        elseif (curr_arg == '-m') or (curr_arg == '--mixed') then
-            insert_cl_config('mode', 'mixed', curr_arg)
-        elseif (curr_arg == '-l') or (curr_arg == '--list') then
-            insert_cl_config('mode', 'list', curr_arg)
-        elseif (curr_arg == '-s') or (curr_arg == '--showall') then
-            insert_cl_config('mode', 'showall', curr_arg)
-
-        -- interaction
-        elseif (curr_arg == '-I') or (curr_arg == '--nointeract') then
-            insert_cl_config('interact_switch', 'false', curr_arg)
-        elseif (curr_arg == '-i') or (curr_arg == '--interact') then
-            insert_cl_config('interact_switch', 'true', curr_arg)
-
-        -- output format
-        elseif (curr_arg == '-M') or (curr_arg == '--machine') then
-            insert_cl_config('machine_switch', 'true', curr_arg)
-
-        -- config
-        elseif curr_arg == '-c' then
-            insert_cl_config(v, nil, curr_arg)
-
-        -- debug
-        elseif (curr_arg == '-d') or (curr_arg == '--debug') then
-            if v == true then v = 'all' end
-            insert_cl_config('debug_list', v, curr_arg)
-        elseif curr_arg == '-D' then
-            insert_cl_config('debug_list', 'all', curr_arg)
-
-        -- verbosity
-        elseif (curr_arg == '-q') or (curr_arg == '--quiet') then
-            insert_cl_config('verbosity_level', C.min_verbosity, curr_arg)
-        elseif (curr_arg == '-v') or (curr_arg == '--verbose') then
-            insert_cl_config('verbosity_level', C.max_verbosity, curr_arg)
+        if #k == 1 then
+            for i, o in pairs(options) do
+                if k == o["short"] then
+                    k = i
+                    goto stop
```

Comment:
`break` should be sufficient here. Please do not use a `goto` statement unless you really have to!